src/GlobalPayments.Api/Terminals/INGENICO/IngenicoController.cs

```diff
@@ -0,0 +1,288 @@
+                var _amount = builder.Amount;
+                var _returnRep = 1;
+                var _paymentMode = 0;
+                var _paymentType = (int)((IngenicoInterface)_device).paymentMethod;
+                var _currencyCode = "826";
+                var _privateData = "EXT0100000";
+                var _immediateAnswer = 0;
+                var _forceOnline = 0;
+                var _extendedData = "0000000000";
+
+                var _cashbackAmount = builder.CashBackAmount;
+                var _authCode = builder.AuthCode;
+                string tableId = builder.TableNumber;
+
+                // Validations
+                _amount = ValidateAmount(_amount);
+                _paymentMode = ValidatePaymentMode(builder.PaymentMode);
+                _currencyCode = ValidateCurrency((string.IsNullOrEmpty(builder.CurrencyCode) ? _currencyCode : builder.CurrencyCode));
+
+                if (!string.IsNullOrEmpty(tableId)) {
+                    bool validateTableId = ValidateTableReference(tableId);
+                    if (validateTableId)
+                        _extendedData = ValidateExtendedData(tableId, builder.ExtendedDataTags);
+                }
+
+                if (!IsObjectNullOrEmpty(_cashbackAmount))
+                    _extendedData = ValidateExtendedData(_cashbackAmount.ToString(), builder.ExtendedDataTags);
+                else if (!string.IsNullOrEmpty(_authCode))
+                    _extendedData = ValidateExtendedData(_authCode, builder.ExtendedDataTags);
+
+
+
+                message = string.Format("{0}{1}{2}{3}{4}{5}{6}A01{7}B01{8}{9}",
+                    _referenceNumber.ToString("00"),
+                    _amount?.ToString("00000000"),
+                    _returnRep,
+                    _paymentMode,
+                    _paymentType,
+                    _currencyCode,
+                    _privateData,
+                    _immediateAnswer,
+                    _forceOnline,
+                    _extendedData);
+            }
+
+            return TerminalUtilities.BuildIngenicoRequest(message, _settings.ConnectionMode);
+        }
+
+        #region Validations
+        private static bool IsObjectNullOrEmpty(object value) {
+            bool response = false;
+            if (value.IsNull() || string.IsNullOrWhiteSpace(value.ToString()))
+                response = true;
+            else response = false;
+
+            return response;
+        }
+        private static bool ValidateTableReference(string value) {
+            bool response = false;
+            if (!string.IsNullOrEmpty(value) && value.Length <= 8)
+                response = true;
+            else throw new BuilderException("Table number must not be less than or equal 0 or greater than 8 numerics.");
+
+            return response;
+        }
+
+        private static int ValidatePaymentMode(PaymentMode _paymentMode) {
+            if (IsObjectNullOrEmpty(_paymentMode)) {
```

Comment:
_paymentMode is not nullable in this method and therefore IsObjectNullOrEmpty will always return false.